app/code/Magento/CustomerGraphQl/Model/Customer/Address/Validator.php

```diff
@@ -0,0 +1,82 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\CustomerGraphQl\Model\Customer\Address;
+
+use Magento\Framework\App\RequestInterface;
+use Magento\Framework\App\RequestInterfaceFactory;
+use Magento\Customer\Model\Metadata\FormFactory;
+use Magento\Customer\Model\Metadata\Form;
+
+class Validator
+{
+    /**
+     * @var RequestInterfaceFactory
+     */
+    private $requestFactory;
+
+    /**
+     * @var FormFactory
+     */
+    private $formFactory;
+
+    /**
+     * @param RequestInterfaceFactory $requestFactory
+     * @param FormFactory $formFactory
+     */
+    public function __construct(
+        RequestInterfaceFactory $requestFactory,
+        FormFactory $formFactory
+    ) {
+        $this->requestFactory = $requestFactory;
+        $this->formFactory = $formFactory;
+    }
+
+    /**
+     * @param array $addressData
+     * @return array|bool
+     */
+    public function validateAddress($addressData)
+    {
+        $request = $this->createNewRequest();
+        $request->setParams($addressData);
+
+        /** @var Form $addressForm */
+        $addressForm = $this->formFactory->create(
```

Comment:
Could you check how it's work in REST api... looks not good when we use something like `form` in HTTP calls responsible area